## Model Evaluation

Evaluate the fine-tuned model on the held-out test set using standard multi-label classification metrics and a per-class classification report.

In [1]:
from pathlib import Path
import sys

PROJECT_ROOT = Path.cwd().parent
sys.path.insert(0, str(PROJECT_ROOT))

In [2]:
import pandas as pd
from ml.evaluate import evaluate_model

metrics, report = evaluate_model()

### Overall Performance

In [5]:
metrics_df = pd.DataFrame(
    metrics.items(),
    columns=["Metric", "Score"],
).round(3)
metrics_df

,Metric,Score
0,Macro F1,0.417
1,Micro F1,0.548
2,Macro ROC AUC,0.804
3,Hamming Loss,0.176


### Per-Class Performance

In [6]:
report_df = pd.DataFrame(report).T.round(3)
report_df

,precision,recall,f1-score,support
angry,0.667,0.500,0.571,4.0
anxious,0.533,0.421,0.471,19.0
bored,0.312,0.714,0.435,7.0
calm,0.500,0.564,0.530,55.0
confused,0.000,0.000,0.000,4.0
excited,0.204,0.289,0.239,38.0
frustrated,0.923,0.571,0.706,21.0
happy,0.622,0.927,0.745,110.0
nostalgic,1.000,0.111,0.200,9.0
proud,0.425,0.667,0.519,51.0


### Summary

The model achieves good overall ranking performance (Macro ROC AUC = 0.804) and reasonable multi-label classification performance (Micro F1 = 0.548). Performance is strongest on common emotions such as *Happy* and *Frustrated*, while less frequent classes like *Confused* and *Nostalgic* remain more difficult to predict.